payjoin/src/receive/mod.rs

```diff
@@ -223,6 +223,11 @@ impl MaybeMixedInputScripts {
     /// Note: mixed spends do not necessarily indicate distinct wallet fingerprints.
     /// This check is intended to prevent some types of wallet fingerprinting.
     pub fn check_no_mixed_input_scripts(self) -> Result<MaybeInputsSeen, RequestError> {
+        // Allow mixed input scripts in payjoin v2
+        if self.params.v == 2 {
+            return Ok(MaybeInputsSeen { psbt: self.psbt, params: self.params });
+        }
+
```

Comment:
I'd rather `receive::v2::MaybeInputsOwned::check_inputs_not_owned` return `MaybeInputsSeen`, skipping `check_no_mixed_input_scripts` in the v2 state machine. We get the same logical result with a code deletion instead of an addition, and don't end up with a sometimes-contradictory function name.

Reply:
I like that. More code deletes!

Reply:
I think the initial suggestion here may be misguided because v2 still needs to handle v1 backwards compatibility. Leaving the state machine as-is, but moving the `self.params.v == 2` conditional to the v2 state machine leaves the opportunity to handle a Mixed Inputs check for v2 without breaking the layer abstraction between v1 and v2 state machines.

Reply:
sounds good to me

Reply:
Worth noting that the BIP78 sender checklist is consistent with your finding

> Verify that the payjoin proposal did not *introduce* mixed input's type.

emphasis my own